### ISSUE TITLE
BLEU Score is not increasing with New Upgrade

### ISSUE THREAD

**Issue by alexwolf22** [tensorflow/tensor2tensor#483]
So yesterday I was training with the transformer_base model, with 1 GPUs and was able to get to BLEU score around 22 after 120k steps and a batch size of 512.
![screen shot 2017-12-21 at 5 20 52 pm](https://user-images.githubusercontent.com/17020909/34277693-de7e69c2-e674-11e7-959e-0f05b31abd10.png)

![screen shot 2017-12-21 at 5 37 46 pm](https://user-images.githubusercontent.com/17020909/34277900-b6afa5a4-e675-11e7-9b2d-90f6426ffebd.png)

I had issues using two GPUs [@266](https://github.com/tensorflow/tensor2tensor/issues/266), so I figured if I updated the library to the latest version, maybe it would work. Unfortunately, it still didn't, but now when I am training the model the BLEU score is extremely low and does not go up, although accuracy and loss gets better(see below). 

![screen shot 2017-12-21 at 5 22 10 pm](https://user-images.githubusercontent.com/17020909/34277557-553740bc-e674-11e7-907c-b41dca1e418c.png)

![screen shot 2017-12-21 at 5 33 47 pm](https://user-images.githubusercontent.com/17020909/34277773-39b7c59a-e675-11e7-9dc5-bfe493a8dd1c.png)

I didn't change any part of my code or the default hyperparameters for the base model I was using, and when I try decoding with this version of the model nothing is outputted. However,  if I decoded near the beginning of this models training it is just jibberish but outputs something. 

I have been scratching my head to what I am doing wrong this time, and have looked into almost everything.

One thing I noticed that the number of parameters for the Transformer Base Model dropped from 61million to 58million with the new update. Is this something that may cause my unresponsive BLEU score or is it something else? Is anyone else having this issue?

**Comment by martinpopel** [comment]
First, note that it was approx_bleu=22, not BLEU=22 (the real BLEU reported in papers tends to be about 1.5 times lower than approx_bleu, see e.g. #436).
As for your question: One possible explanation is that he training has diverged and you need to increase learning_rate_warmup_steps, e.g. using `--hparams="learning_rate_warmup_steps=32000"`.
That said, I have no idea why upgrading to a new version causes the problem (BTW: you should write which exact version it is when reporting an issue).

**Comment by alexwolf22** [comment]
I normally post previous update version but didn't know at the time when I posted what it was, so I didn't sorry. I was using version 1.2.4. 

I am training the model with a higher warmup step now. Seems to not diverge as quickly. Thanks for the help.

**Comment by rsepassi** [comment]
Thanks @martinpopel
